src/Framework/AspNetCoreAnalyzers/test/RouteEmbeddedLanguage/RoutePatternAnalyzerTests.cs

```diff
@@ -512,4 +521,38 @@ public object TestAction(int id)
         // Assert
         Assert.Empty(diagnostics);
     }
+
+    [Fact]
+    public async Task ConcatString_PerformanceTest()
+    {
+        // Arrange
+        var fileText = File.ReadAllText(System.IO.Path.Combine("RouteEmbeddedLanguage", "Class1.cs.txt"));
```

Comment:
>File.ReadAllText(System.IO.Path.Combine("RouteEmbeddedLanguage", "Class1.cs.txt")) [](http://example.com/codeflow?start=23&length=82)

Is the actual string content important? If not, perhaps generate the source in the test rather than checking it in.
```csharp
var builder = new System.Text.StringBuilder();
builder.AppendLine("""
    class Program
    {
        static void Main() { }
        static readonly string _s =
    """);
for (int i = 0; i < 2000; i++)
    builder.AppendLine("        \"abc\" +");
builder.AppendLine("""
            "";
    }
    """);
var source = TestSource.Read(builder.ToString());
```